crate/test_server/src/test_server.rs

```diff
@@ -160,60 +205,84 @@ async fn wait_for_server_to_start(kms_client: &KmsClient) -> Result<(), ClientEr
-                    port,
-                    https_p12_file: Some(
-                        root_dir.join("certificates/server/kmserver.acme.com.p12"),
-                    ),
-                    https_p12_password: Some("password".to_string()),
-                    authority_cert_file: Some(root_dir.join("certificates/server/ca.crt")),
-                    ..HttpConfig::default()
-                }
-            } else {
-                HttpConfig {
-                    port,
-                    https_p12_file: Some(
-                        root_dir.join("certificates/server/kmserver.acme.com.p12"),
-                    ),
-                    https_p12_password: Some("password".to_string()),
-                    ..HttpConfig::default()
-                }
+
+    if use_https {
+        if use_client_cert {
+            HttpConfig {
+                port,
+                https_p12_file: Some(root_dir.join("certificates/server/kmserver.acme.com.p12")),
+                https_p12_password: Some("password".to_string()),
+                authority_cert_file: Some(root_dir.join("certificates/server/ca.crt")),
+                api_token_id,
+                ..HttpConfig::default()
             }
         } else {
             HttpConfig {
                 port,
+                https_p12_file: Some(root_dir.join("certificates/server/kmserver.acme.com.p12")),
+                https_p12_password: Some("password".to_string()),
+                api_token_id,
                 ..HttpConfig::default()
             }
+        }
+    } else {
+        HttpConfig {
+            port,
+            api_token_id,
+            ..HttpConfig::default()
+        }
+    }
+}
+
+fn generate_server_params(
+    db_config: DBConfig,
+    port: u16,
+    authentication_options: &AuthenticationOptions,
+) -> Result<ServerParams, ClientError> {
+    // Configure the server
+    let clap_config = ClapConfig {
+        auth: if authentication_options.use_jwt_token {
+            get_auth0_jwt_config()
+        } else {
+            JwtAuthConfig::default()
         },
+        db: db_config,
+        http: generate_http_config(
+            port,
+            authentication_options.use_https,
+            authentication_options.use_client_cert,
+            authentication_options.api_token_id.clone(),
+        ),
         ..ClapConfig::default()
     };
     ServerParams::try_from(clap_config)
         .map_err(|e| ClientError::Default(format!("failed initializing the server config: {e}")))
 }
 
-fn generate_owner_conf(server_params: &ServerParams) -> Result<(String, ClientConf), ClientError> {
+fn set_access_token(server_params: &ServerParams, api_token: Option<String>) -> Option<String> {
+    if server_params.identity_provider_configurations.is_some() {
+        println!("Setting access token for JWT: {AUTH0_TOKEN:?}");
+        Some(AUTH0_TOKEN.to_string())
+    } else if api_token.is_some() {
+        println!("Setting access token for API: {api_token:?}");
```

Comment:
no more `println!` no? `tracing::debug!` instead ?

Reply:
done in 1406128fcd3be5c89c03570cd3cc25eecaddad17